Kernel/CommandLine.cpp

```diff
@@ -108,6 +108,9 @@ UNMAP_AFTER_INIT bool CommandLine::is_ide_enabled() const
 
 UNMAP_AFTER_INIT bool CommandLine::is_smp_enabled() const
 {
+    // Note: We can't enable SMP mode without enabling the IOAPIC.
+    if (!is_ioapic_enabled())
```

Comment:
Should we log a message?

Reply:
For every call to `is_smp_enabled()`? Probably not. Maybe at the call site (the caller should be responsible to print it), right?

Reply:
There is only one call, in `InterruptManagement::initialize()`. I was just thinking it would be nice to inform the user that we have deiced we can't enable SMP, instead of just blindly doing it an continuing. Sure, this function probably isn't the best place to do that.